### DISCUSSION TITLE
[Feature] Have the Slideshow "Shuffle" be non-repeating until after all album items have been viewed.

### DISCUSSION THREAD

**Discussion by mnjm97** [immich-app/immich#9016] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

could the random play be set to non-repeating until after all photos/videos have been played before looping again to a new random set for the slideshow.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by UnoSD** [comment]
I was about to open the same; not sure why it keeps showing the same assets over and over again. Possibly alternates years and repeats the same image if a year has always the same? the api search/random seems to work well and not having this issue when picking one asset at the time.

**Reply by UnoSD** [reply]
On a side note, I just published this small script to get a more "random" slideshow: https://github.com/UnoSD/ImmichMpvSlideshow/ happy to send a PR if it's worth adding to the Community Projects despite being tiny.

**Reply by SimJoSt** [reply]
The shuffle mode should have fewer repeats according to the changelog of the last versions.

**Reply by UnoSD** [reply]
@SimJoSt I can confirm that it does for me
